Source/Core/DolphinQt2/Debugger/MemoryWidget.cpp

```diff
@@ -0,0 +1,575 @@
+  search_layout->addWidget(m_find_previous);
+  search_layout->addWidget(m_result_label);
+
+  // Data Type
+  auto* datatype_group = new QGroupBox(tr("Data Type"));
+  auto* datatype_layout = new QVBoxLayout;
+  datatype_group->setLayout(datatype_layout);
+
+  m_type_u8 = new QRadioButton(tr("U&8"));
+  m_type_u16 = new QRadioButton(tr("U&16"));
+  m_type_u32 = new QRadioButton(tr("U&32"));
+  m_type_ascii = new QRadioButton(tr("ASCII"));
+  m_type_float = new QRadioButton(tr("Float"));
+
+  datatype_layout->addWidget(m_type_u8);
+  datatype_layout->addWidget(m_type_u16);
+  datatype_layout->addWidget(m_type_u32);
+  datatype_layout->addWidget(m_type_ascii);
+  datatype_layout->addWidget(m_type_float);
+
+  // MBP options
+  auto* bp_group = new QGroupBox(tr("Memory breakpoint options"));
+  auto* bp_layout = new QVBoxLayout;
+  bp_group->setLayout(bp_layout);
+
+  m_bp_read_write = new QRadioButton(tr("Read and Write"));
+  m_bp_read_only = new QRadioButton(tr("Read only"));
+  m_bp_write_only = new QRadioButton(tr("Write only"));
+  m_bp_log_check = new QCheckBox(tr("Log"));
+
+  bp_layout->addWidget(m_bp_read_write);
+  bp_layout->addWidget(m_bp_read_only);
+  bp_layout->addWidget(m_bp_write_only);
+  bp_layout->addWidget(m_bp_log_check);
+
+  // Sidebar
+  auto* sidebar = new QWidget;
+  auto* sidebar_layout = new QVBoxLayout;
+  sidebar->setLayout(sidebar_layout);
+
+  sidebar_layout->addWidget(m_search_address);
+  sidebar_layout->addWidget(m_data_edit);
+  sidebar_layout->addWidget(m_find_ascii);
+  sidebar_layout->addWidget(m_find_hex);
+  sidebar_layout->addWidget(m_set_value);
+  sidebar_layout->addItem(new QSpacerItem(1, 32));
+  sidebar_layout->addWidget(m_dump_mram);
+  sidebar_layout->addWidget(m_dump_exram);
+  sidebar_layout->addWidget(m_dump_fake_vmem);
+  sidebar_layout->addWidget(search_group);
+  sidebar_layout->addWidget(datatype_group);
+  sidebar_layout->addWidget(bp_group);
+
+  // Splitter
+  m_splitter = new QSplitter(Qt::Horizontal);
+
+  auto* sidebar_scroll = new QScrollArea;
+  sidebar_scroll->setWidget(sidebar);
+  sidebar_scroll->setWidgetResizable(true);
+  sidebar_scroll->setMinimumWidth(250);
+
+  m_memory_view = new MemoryViewWidget;
+
+  m_splitter->addWidget(m_memory_view);
+  m_splitter->addWidget(sidebar_scroll);
+
+  layout->addWidget(m_splitter);
+
+  auto* widget = new QWidget;
+  widget->setLayout(layout);
+  setWidget(widget);
+}
+
+void MemoryWidget::ConnectWidgets()
+{
+  connect(m_search_address, &QLineEdit::textChanged, this, &MemoryWidget::OnSearchAddress);
+
+  connect(m_data_edit, &QLineEdit::textChanged, this, &MemoryWidget::ValidateSearchValue);
+  connect(m_find_ascii, &QRadioButton::toggled, this, &MemoryWidget::ValidateSearchValue);
+  connect(m_find_hex, &QRadioButton::toggled, this, &MemoryWidget::ValidateSearchValue);
+
+  connect(m_set_value, &QPushButton::pressed, this, &MemoryWidget::OnSetValue);
+
+  connect(m_dump_mram, &QPushButton::pressed, this, &MemoryWidget::OnDumpMRAM);
```

Comment:
Some of these slots could be defined inline as lambdas, e.g.
```cpp
connect(m_dump_mram, &QPushButton::pressed, this, [this]() { DumpArray(File::GetUserPath(F_RAMDUMP_IDX), Memory::m_pRAM, Memory::REALRAM_SIZE); } );
````

Reply:
Sure but it is a bit clearer with separate functions IMO.